Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1220,18 +1222,28 @@ private boolean readAndStripParam(Uri data, Activity activity) {
             // Capture the intent URI and extra for analytics in case started by external intents such as  google app search
             try {
                 if (data != null) {
+                    boolean foundSchemeMatch;
+                    boolean skipThisPath = false;
                     if (externalUriWhiteList_.size() > 0) {
-                        String externalIntentScheme = Uri.parse(data.toString()).getScheme();
-                        if (externalIntentScheme != null && externalUriWhiteList_.contains(externalIntentScheme)) {
-                            sessionReferredLink_ = data.toString();
-                            prefHelper_.setExternalIntentUri(data.toString());
-                        }
+                        foundSchemeMatch = externalUriWhiteList_.contains(data.getScheme());
                     } else {
+                        foundSchemeMatch = true;
+                    }
+                    if (skipExternalUriPaths_.size() > 0) {
+                        for (String path : skipExternalUriPaths_) {
+                            String externalPath = data.getPath();
```

Comment:
@aaronhe42 Sorry for the confusion. We have released 2.4.4 which has the option to skip collecting external intent data on specified hot name. Could you please try that.